packages/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx

```diff
@@ -0,0 +1,615 @@
+      isBeakVisible: false,
+      gapSpace: 15,
+      color: this.state.color,
+      legend: this.state.selectedLegendTitle,
+      XValue: this.state.xCalloutValue,
+      YValue: this.state.yCalloutValue ? this.state.yCalloutValue : this.state.dataForHoverCard,
+      onDismiss: this._closeCallout,
+      preventDismissOnLostFocus: true,
+      ...this.props.calloutProps,
+      ...getAccessibleDataObject(this.state.callOutAccessibilityData),
+    };
+    const tickParams = {
+      tickValues: this.props.tickValues,
+      tickFormat: this.props.tickFormat,
+    };
+    return (
+      <CartesianChart
+        {...this.props}
+        points={this._points}
+        chartType={ChartTypes.HorizontalBarChartWithAxis}
+        xAxisType={this._xAxisType}
+        yAxisType={this._yAxisType}
+        stringDatasetForYAxisDomain={this._yAxisLabels}
+        calloutProps={calloutProps}
+        tickParams={tickParams}
+        legendBars={legendBars}
+        barwidth={this._barWidth}
+        focusZoneDirection={FocusZoneDirection.vertical}
+        customizedCallout={this._getCustomizedCallout()}
+        getmargins={this._getMargins}
+        getGraphData={this._getGraphData}
+        getAxisData={this._getAxisData}
+        onChartMouseLeave={this._handleChartMouseLeave}
+        /* eslint-disable react/jsx-no-bind */
+        // eslint-disable-next-line react/no-children-prop
+        children={(props: IChildProps) => {
+          return (
+            <>
+              <g>{this._bars}</g>
+            </>
+          );
+        }}
+      />
+    );
+  }
+
+  private _adjustProps(): void {
+    this._points = this.props.data || [];
+    this._barWidth = this.props.barWidth || 32;
+    const { palette } = this.props.theme!;
+    this._colors = this.props.colors || [palette.blueLight, palette.blue, palette.blueMid, palette.blueDark];
+  }
+
+  private _getMargins = (margins: IMargins) => {
+    this.margins = margins;
+  };
+
+  private _renderContentForOnlyBars = (props: IHorizontalBarChartWithAxisDataPoint): JSX.Element => {
+    const { useSingleColor = false } = this.props;
+    return (
+      <>
+        <ChartHoverCard
+          XValue={props.xAxisCalloutData || props.x.toString()}
+          Legend={props.legend}
+          YValue={props.yAxisCalloutData || props.y}
+          color={!useSingleColor && props.color ? props.color : this._createColors()(props.x)}
+          culture={this.props.culture}
+        />
+      </>
+    );
+  };
+
+  private _renderCallout = (props?: IHorizontalBarChartWithAxisDataPoint): JSX.Element | null => {
+    return props ? this._renderContentForOnlyBars(props) : null;
+  };
+
+  private _getCustomizedCallout = () => {
+    return this.props.onRenderCalloutPerDataPoint
+      ? this.props.onRenderCalloutPerDataPoint(this.state.dataPointCalloutProps, this._renderCallout)
+      : null;
+  };
+
+  private _getGraphData = (
+    xScale: StringAxis,
+    yScale: NumericAxis,
```

Comment:
Y can be numeric or string both. Did we decide to implement numeric variant only.

Reply:
Changed